online-shop/src/app/product-details/product-details.component.ts

```diff
@@ -1,22 +1,34 @@
-import {Component, OnInit} from '@angular/core';
-import {ActivatedRoute} from "@angular/router";
-import {Product} from "../../product";
-import {products} from "../../product";
-
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import { Product } from '../../product';
+import {ProductService} from "../product.service";
 
 @Component({
   selector: 'app-product-details',
   templateUrl: './product-details.component.html',
-  styleUrls: ['./product-details.component.scss']
+  styleUrls: ['./product-details.component.scss'],
 })
 export class ProductDetailsComponent implements OnInit {
   id = 0;
-  product!: Product
-constructor(
-  private route: ActivatedRoute,
-) {}
+  product!: Product;
+  constructor(private route: ActivatedRoute, private productService: ProductService) {}
+
+
```

Comment:
as you write code, i know the need of having a clean space to work on, but you have to get used to:
-run over the changed files look for leftover code, unused vars, comments that are not needed, spaces,
-format the code
-run linter